toast-tk-web-agent/src/main/java/io/toast/tk/agent/ui/panels/AbstractPanel.java

```diff
@@ -0,0 +1,177 @@
+		this.strkey = strkey;
+		
+		this.notvalidImage = PanelHelper.createImage(this,"picto-non-valide.png");
+		this.validImage = PanelHelper.createImage(this,"picto-valide.png");
+		this.toastLogo = PanelHelper.createImage(this,"ToastLogo_24.png");
+				
+		this.buildContentPanel();
+		
+		if(dim == null) {
+			dim = getPreferredSize();
+		}
+		else {
+			setPreferredSize(dim);
+		}		
+	}
+	
+	public String getTextValue() {
+		return this.textField.getText();
+	}
+
+	private JTextField createBasicTextPanel() {
+		JTextField textField = new JTextField(render(properties.getProperty(strkey)));
+		Dimension prefDim = new Dimension(textField.getPreferredSize().width, 30);
+		Dimension moyDim = new Dimension(textField.getMaximumSize().width, 40);
+		textField.setPreferredSize(prefDim);
+        textField.setMaximumSize(moyDim);
+        textField.setFont(PanelHelper.FONT_TEXT);
+		textField.setLayout(new BorderLayout());
+		textField.setColumns(30);
+		textField.setAlignmentX(Component.LEFT_ALIGNMENT);	
+		textField.setAlignmentY(Component.CENTER_ALIGNMENT);
+		textField.addKeyListener(new KeyListener() {
+			@Override
+			public void keyPressed(KeyEvent a) {
+				if (a.getKeyCode() == KeyEvent.VK_ENTER) {
+					try {
+						testIconValid(false);
+					} catch (IOException e) {
+						LOG.error(e.getMessage(), e);
+					}
+				}
+			}
+
+			@Override
+			public void keyReleased(KeyEvent e) {
+			}
+
+			@Override
+			public void keyTyped(KeyEvent e) {
+			}
+		});
+		
+		return textField;
+	}
+
+	public abstract void testIconValid(boolean retry) throws IOException;
+
+	public JLabel createIconValid() {
+		JLabel iconValid = new JLabel(new ImageIcon(this.validImage));
+		return iconValid;
+	}
+	
+	public JLabel createIconNotValid() {
+		JLabel iconNotValid = new JLabel(new ImageIcon(this.notvalidImage));
+		return iconNotValid;
+	}
+
+	protected abstract JComponent buildPanel() throws IOException;
+
+	private void buildContentPanel() throws IOException{	
+		this.add(Box.createHorizontalGlue());
+		this.setLayout(new BoxLayout(this, BoxLayout.PAGE_AXIS));
+		this.setAlignmentX(Component.CENTER_ALIGNMENT);
+		this.setAlignmentY(Component.CENTER_ALIGNMENT);
+		this.setBackground(Color.white);
+		
+		JPanel textButtonPanel = PanelHelper.createBasicPanel(BoxLayout.LINE_AXIS);
+
+		textField = createBasicTextPanel();
+		iconPanel = PanelHelper.createBasicPanel();
+		iconValid = createIconValid();
+		iconNotValid = createIconNotValid();
+
+		JComponent component = buildPanel();
+		if(component != null){
+			textButtonPanel.add(component);
+		}
+
+		textButtonPanel.add(textField);
+		textButtonPanel.add(iconPanel);
+
+		textButtonPanel.setSize(new Dimension(textButtonPanel.getSize().width, 50));
+		textButtonPanel.setMaximumSize(new Dimension(textButtonPanel.getMaximumSize().width, 50));
+		
+		this.add(textButtonPanel);
+		this.add(errorLabel);
+	}
+	
+	protected JLabel buildErrorLabel(JPanel iconPanel, EnumError testValue) throws IOException{
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'java.io.IOException', as it cannot be thrown from method's body. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)